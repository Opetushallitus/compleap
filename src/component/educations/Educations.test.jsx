import React from 'react'
import createRendererWithTheme from 'test/util/createRendererWithTheme'

const mockState = formState => jest.doMock('state/state', () => {
  const Atom = require('bacon.atom')
  const React = require('react')

  const state = Atom({
    context: {
      user: {
        language: 'fi'
      },
      interests: {
        data: [],
        error: undefined
      },
      education: {
        data: {
          unverifiedEducations: [{
            id: 'd17ebd27-c42c-416a-954c-2eb42f9436ed',
            level: { id: '1' }
          }, {
            id: 'a24d7a2e-2922-4f60-9d12-c1285ead6ddd',
            level: { id: '2' },
            specifier: { id: '0214' },
            code: 'koulutus_321101'
          }],
          selection: undefined
        }
      }
    },
    value: {
      profile: {
        education: {
          unverifiedEducation: formState
        }
      }
    }
  })

  return {
    Context: React.createContext(state),
    state: Atom(state)
  }
})

describe('Education', () => {
  beforeAll(() => {
    global.console = Object.assign({}, global.console, { warn: jest.fn() }) // Suppress logging
  })

  beforeEach(() => {
    jest.clearAllMocks()
    jest.mock('component/educations/unverified-educations-list/fragment/Competences', () => () => null)
  })

  it('should show added educations', () => {
    let Education
    mockState('formCollapsed')
    jest.isolateModules(() => { Education = require('./Educations').default })

    const renderedJSON = createRendererWithTheme(<Education/>).toJSON()
    expect(renderedJSON).toMatchSnapshot()
  })
})
