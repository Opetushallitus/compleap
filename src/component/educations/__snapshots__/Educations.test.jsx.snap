// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education should show added educations 1`] = `
Array [
  <h1
    className="sc-bdVaJa hqCxXu"
  >
    Koulutukset
  </h1>,
  <h2
    className="sc-bwzfXH krgDD"
  >
    Ulkomailla suoritetut tutkinnot
  </h2>,
  <div
    className="sc-ifAKCX dOlJFM"
  >
    <div>
      <h3
        className="sc-htpNat cxoWVD"
      >
        Perusopetus
      </h3>
    </div>
    <div>
      <button
        className="sc-kpOJdX sc-jKJlTe hXbFZE"
        onClick={[Function]}
      >
        Poista tutkinto
      </button>
    </div>
  </div>,
  <div
    className="sc-ifAKCX dOlJFM"
  >
    <div>
      <h3
        className="sc-htpNat cxoWVD"
      >
        Ammatillinen tutkinto
      </h3>
      <p
        className="sc-hMqMXs dtCVTd"
      >
        Käsityöalat
      </p>
    </div>
    <div>
      <button
        className="sc-kpOJdX sc-jKJlTe hXbFZE"
        onClick={[Function]}
      >
        Poista tutkinto
      </button>
    </div>
  </div>,
  <button
    className="sc-kpOJdX sc-ckVGcZ fbtVZP"
    onClick={[Function]}
  >
    Lisää muualla kuin Suomessa suoritettu tutkinto
  </button>,
]
`;
